drivers/volume/portworx/portworx.go

```diff
@@ -330,19 +348,33 @@ func (d *portworx) StopDriver(n node.Node) error {
 	return d.schedOps.DisableOnNode(n)
 }
 
-func (d *portworx) WaitStart(n node.Node) error {
-	var err error
-	// Wait for Portworx to become usable.
+func (d *portworx) getClusterOnStart() (api.Cluster, error) {
 	t := func() (interface{}, error) {
-		clusterManager := d.getClusterManager()
-		if status, _ := clusterManager.NodeStatus(); status != api.Status_STATUS_OK {
-			return "", &ErrFailedToWaitForPx{
-				Node:  n,
-				Cause: fmt.Sprintf("px cluster is still not up. Status: %v", status),
+		cluster, err := d.getClusterManager().Enumerate()
+		if err != nil {
+			return nil, err
+		}
+		if cluster.Status != api.Status_STATUS_OK {
+			return nil, &ErrFailedToWaitForPx{
+				Cause: fmt.Sprintf("px cluster is still not up. Status: %v", cluster.Status),
 			}
 		}
 
-		pxNode, err := clusterManager.Inspect(n.Name)
+		return cluster, nil
+	}
+
+	cluster, err := task.DoRetryWithTimeout(t, 2*time.Minute, 10*time.Second)
+	if err != nil {
+		return api.Cluster{}, err
```

Comment:
In an error case, it's better to return a nil object than a default one.

Reply:
it was an object, hence could not return a nil
but changed it to a reference